conans/client/cmake.py

```diff
@@ -122,10 +122,26 @@ def flags(self):
 
     @property
     def runtime(self):
+        runtime = ""
+        deprecation_warn = False
+        # Check if exists the old compiler.runtime
         try:
             runtime = self._settings.compiler.runtime
+            if runtime.value:
+                deprecation_warn = True
         except ConanException:
-            return ""
+            pass
+
+        try:
+            runtime = self._settings.msvcrt
+            if runtime.value:
+                deprecation_warn = False
+        except ConanException:
+            pass
+
+        ret = ""
         if runtime:
-            return "-DCONAN_LINK_RUNTIME=/%s" % runtime
-        return ""
+            ret = "-DCONAN_LINK_RUNTIME=/%s" % runtime
+            if deprecation_warn:
+                ret += " -DRUNTIME_DEPRECATION_WARNING=ON"
```

Comment:
Why this flag to generate the warning in cmake (and not other generators)? Why not issuing the warning right here?